libbeat/docs/queueconfig.asciidoc

```diff
@@ -76,3 +77,147 @@ will be immediately available for consumption.
+    flush.timeout: 5s
+    flush.events: 1024
+------------------------------------------------------------------------------
+
+[float]
+==== Configuration options
+
+You can specify the following options in the `queue.spool` section of the
++{beatname_lc}.yml+ config file:
+
+[float]
+===== `file.path`
+
+The spool file path. The file is created on startup, if it does not exist.
+
+The default value is "${path.data}/spool.dat".
+
+===== `file.permissions`
+
+The file permissions. The permissions are applied when the file is
+created. In case the file already exists, The file permissions are compared
+with `file.permissions`. The spool file is not opened if the actual file
+permissions are more permissive then configured.
+
+The default value is 0600.
+
+
+===== `file.size`
+
+Spool file size. 
+
+The default value is 100 MiB.
+
+Note: The size should be much larger then the expected event sizes
+and write buffer size. Otherwise the queue will block, because it has not
+enough space.
+
+Note: The file size can not be changed once the file has been generated. This
+limitation will be removed in the future.
+
+
+===== `file.page_size`
+
+The files page size.
+
+The spool file is split into multiple pages of `page_size`. All write
+operations write a complete page.
+
+The default value is 4096 (4KiB).
+
+Note: This setting should match the file systems minimum block size. If the
+`page_size` is not a multiple of the file systems block size, the file system
+might create additional read operations on writes.
+
+Note: The page size is only set on file creation time. It can not be changed
+afterwards.
+
+
+===== `file.prealloc`
+
+If `prealloc` is set to `true`, truncate is used to reserve the space up to
+`file.size`. This setting is only used when the file is created.
+
+The file will dynamically grow, if `prealloc` is set to false. The spool
+blocks, if `prealloc` is `false` and the system is out of disk space.
+
+The default value is `true`.
+
+===== `write.buffer_size`
+
+The write buffer size. The write buffer is flushed, once the buffer size is exceeded.
+
+Very big events are allowed to be bigger then the configured buffer size. But
+the write buffer will be flushed right after the event has been serialized.
+
+The default value is 1MiB.
+
+===== `write.codec`
+
+The event encoding used for serialized events. Valid values are `json` and `cbor`.
+
+The default value is `cbor`.
+
+===== `write.flush.timeout`
+
+Maximum wait time of the oldest event in the write buffer. If set to 0, the
+write buffer will only be flushed once `write.flush.events` or `write.buffer_size` is fulfilled.
+
+The default value is 1s.
+
+===== `write.flush.events`
+
+Number of buffered events. The write buffer is flushed once the limit is reached.
+
+The default value is 16384.
+
+===== `read.flush.timeout`
+
+If configured, the `read.flush.timeout` ensures the spool will wait for more
+events to be flushed to the spool file, If the number of available events is
```

Comment:
A couple of typos. Should be:

...if the number of available events is less than the output's `bulk_max_size`...

Reply:
I have a hard time writing docs on this setting. Did rewrite complete section. Hope it's somewhat better now.